File: HelpSoftware/Help/EF/ViewHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Help
{
  public class ViewHistory
  {
    [Key]
    public int ID_ViewHistory { get; set; }

    public Article ViewedArticle { get; set; }

    public User ViewedUser { get; set; }

    public DateTime AccessTime { get; set; }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/ViewHistory.ts
import { Article, User } from "./EF";

export class ViewHistory {
  ID_ViewHistory: number;
  ViewedArticle: Article;
  ViewedUser: User;
  AccessTime: Date;

  deserialize(input) {
    this.ID_ViewHistory = input.ID_ViewHistory;
    this.ViewedArticle = new Article().deserialize(input.ViewedArticle);
    this.ViewedUser = new User().deserialize(input.ViewedUser);
    this.AccessTime = input.AccessTime;
    return this;
  }
}